reconcilers/resource.go

```diff
@@ -74,6 +74,10 @@ type ResourceReconciler[Type client.Object] struct {
 	// conflicts. Finalizers and events are still actionable.
 	SkipStatusUpdate bool
 
+	// AllowStatusUpdateDuringDeletion when true, the rsource's status will be updated even
+	// when the resource is marked for deletion.
+	AllowStatusUpdateDuringDeletion bool
```

Comment:
```suggestion
	SyncStatusDuringFinalization bool
```

In the `SyncReconciler` we have a field named [`SyncDuringFinalization`](https://pkg.go.dev/reconciler.io/runtime/reconcilers#SyncReconciler.SyncDuringFinalization).  Suggest aligning with that.

Reply:
Updated